var/spack/repos/builtin/packages/lazyten/package.py

```diff
@@ -0,0 +1,101 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+
+
+class Lazyten(CMakePackage):
+    """Lightweight linear algebra library based on lazy matrices"""
+
+    maintainers = ['mfherbst']
+
+    homepage = "http://lazyten.org"
+    url      = "https://github.com/lazyten/lazyten/archive/v0.4.1.tar.gz"
+
+    version('0.4.1', 'd06f7996144e1bf1b0aee82c2af36e83')
+    version("develop", git="https://github.com/lazyten/lazyten.git",
+            branch="master")
+
+    variant("build_type", default="DebugRelease",
+            description="The build type to build",
+            values=("Debug", "Release", "DebugRelease"))
+    variant("examples", default=False,
+            description="Compile examples")
+    variant("arpack", default=True,
+            description="Build with Arpack support")
+
+    # Build dependencies
+    depends_on("cmake@3.0:", type="build")
+
+    # Only builds on clang > 3.5 and gcc > 4.8
+    conflicts("%intel", msg="lazyten only builds with gcc and clang")
+    conflicts("%gcc@:4.8")
+    conflicts("%clang@:3.5")
+
+    depends_on("krims@develop", when="@develop")
+    depends_on("krims@0.2.1", when="@0.4.1")
+
+    depends_on("blas")
+    depends_on("lapack")
+    depends_on("armadillo@4.000:")
```

Comment:
Why `000`? Isn't `4` enough?

Reply:
Armadillos documentation always uses version numbers like `n.xxx`, so I thought I'd adhere to that convention. I guess `4:` is easier to read.

Reply:
in spack `4:` will be anything above `4`, including `4.1`. So that should work indeed